src/playbacks/html5_video/html5_video.js

```diff
@@ -426,6 +447,96 @@ export default class HTML5Video extends Playback {
     this.trigger(Events.PLAYBACK_READY, this.name)
   }
 
+  _checkForClosedCaptions() {
+    // Check if CC available only if current playback is HTML5Video
+    if (this.isHTML5Video && !this._ccIsSetup) {
+      if (this.hasClosedCaptionsTracks) {
+        this.trigger(Events.PLAYBACK_SUBTITLE_AVAILABLE)
+        const trackId = this.closedCaptionsTrackId
+        this.closedCaptionsTrackId = trackId
+        this.handleTextTrackChange = this._handleTextTrackChange.bind(this)
+        this.el.textTracks.addEventListener('change', this.handleTextTrackChange)
+      }
+      this._ccIsSetup = true
+    }
+  }
+
+  _handleTextTrackChange() {
+    let tracks = this.closedCaptionsTracks
+    let track = tracks.find((track) => {
+      if (track.track.mode === 'showing') {
+        return track
+      }
+    }) || {id: -1}
+    if (this._ccTrackId !== track.id) {
+      this._ccTrackId = track.id
+      this.trigger(Events.PLAYBACK_SUBTITLE_CHANGED, {
+        id: track.id
+      })
+    }
+  }
+
+  get isHTML5Video() {
+    return this.name === HTML5Video.prototype.name
+  }
+
+  get closedCaptionsTracks() {
+    let id = 0
+    let newId = () => { return id++ }
+    let onlySubtitles = (track) => { return track.kind === 'subtitles' }
+    let makeTrack = (track) => { return {id: newId(), name: track.label, track: track} }
+    let textTracks = this.el.textTracks ? Array.from(this.el.textTracks) : []
+
+    return textTracks.filter(onlySubtitles).map(makeTrack)
+  }
+
+  get closedCaptionsTrackId() {
+    return this._ccTrackId
+  }
+
+  set closedCaptionsTrackId(trackId) {
+    if (!isNumber(trackId)) {
+      return
+    }
+
+    let tracks = this.closedCaptionsTracks
+    let showingTrack
+
+    if (trackId !== -1) {
+      // Get track to show
+      showingTrack = tracks.find((track) => {
+        if (track.id == trackId) {
+          return track
+        }
+      })
+    }
+
+    // Ensure track is not already displayed
+    if (showingTrack && showingTrack.track.mode === 'showing') {
+      return
+    }
+
+    // Since it is possible to display multiple tracks,
+    // ensure that all tracks are hidden.
+    tracks
+      .filter((track) => { return track.track.mode !== 'hidden' })
+      .forEach((track) => { track.track.mode = 'hidden' })
+
+    // Display track
+    if (showingTrack) {
+      showingTrack.track.mode = 'showing'
+    } else if (trackId !== -1) {
```

Comment:
Since `trackId` never get changed during this function life cycle I'm in favor of moving this statement up above like if it was a [precondition](https://github.com/google/guava/wiki/PreconditionsExplained) to run this functions? what do you think? It's like, if the track is different from `-1` then we should be doing nothing, I'm positively saying this because we didn't change its value.

Should be mark all the tracks as hidden even if the trackId is different from -1?

Does `-1` means selected?